dist/platforms/ubuntu/entrypoint.sh

```diff
@@ -1,47 +1,63 @@
 #!/usr/bin/env bash
 
-#
-# Create directory for license activation
-#
-
-ACTIVATE_LICENSE_PATH="$GITHUB_WORKSPACE/_activate-license~"
-mkdir -p "$ACTIVATE_LICENSE_PATH"
-
-#
-# Run steps
-#
-source /steps/set_extra_git_configs.sh
-source /steps/set_gitcredential.sh
-source /steps/activate.sh
-source /steps/build.sh
-source /steps/return_license.sh
-
-#
-# Remove license activation directory
-#
-
-rm -r "$ACTIVATE_LICENSE_PATH"
-chmod -R 777 "/BlankProject"
-
-#
-# Instructions for debugging
-#
-
-if [[ $BUILD_EXIT_CODE -gt 0 ]]; then
-echo ""
-echo "###########################"
-echo "#         Failure         #"
-echo "###########################"
-echo ""
-echo "Please note that the exit code is not very descriptive."
-echo "Most likely it will not help you solve the issue."
-echo ""
-echo "To find the reason for failure: please search for errors in the log above."
-echo ""
-fi;
-
-#
-# Exit with code from the build step.
-#
-
-exit $BUILD_EXIT_CODE
+fullProjectPath="$GITHUB_WORKSPACE/$PROJECT_PATH"
+
+# Get host user/group info so we create files with the correct ownership
+USERNAME=$(stat -c '%U' "$fullProjectPath")
+USERID=$(stat -c '%u' "$fullProjectPath")
+GROUPNAME=$(stat -c '%G' "$fullProjectPath")
+GROUPID=$(stat -c '%g' "$fullProjectPath")
+
+groupadd -g $GROUPID $GROUPNAME
+useradd -u $USERID -g $GROUPID $USERNAME
+usermod -aG $GROUPNAME $USERNAME
+mkdir -p "/home/$USERNAME"
+chown $USERNAME:$GROUPNAME "/home/$USERNAME"
+
+# Normally need root permissions to access when using su
+chmod 777 /dev/stdout
+chmod 777 /dev/stderr
```

Comment:
We used to have `chmod -R 777 "/BlankProject"` here. I don't have much context, but there might be security concerns about these two linces. Why is-this necessary exactly? Could it be more specific or behind a condition?

Reply:
The chmod didn't seem to fix the issue completely as a cancelled build wouldn't reach the line and cause runner agents to fail to delete the files. This PR is an overarching fix for permissions such that in the future if new files or folders are generated they will have the right ownership and we won't need to keep chasing these issues down. I'm not concerned on the security front because this is within the container and previously it was running as root already so any semblance of security was already out the window as the user could do anything they wanted to. These needed to be added so the non root user can write to stdout which is baked into the docker images for xvfb to work hence doing the workaround here. I will however move this within the if statement as it is only needed for non-root users.